server/src-lib/Hasura/Events/HTTP.hs

```diff
@@ -0,0 +1,362 @@
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE MultiWayIf            #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TemplateHaskell       #-}
+
+module Hasura.Events.HTTP
+  ( HTTP(..)
+  , HTTPSessionMgr(..)
+  , mkHTTP
+  , mkAnyHTTPPost
+  , mkHTTPMaybe
+  , HTTPErr(..)
+  , runHTTP
+  , runInsecureHTTP
+  , default2xxParser
+  , noBody2xxParser
+  , defaultRetryPolicy
+  , defaultRetryFn
+  , defaultParser
+  , defaultParserMaybe
+  , isNetworkError
+  , isNetworkErrorHC
+  , HLogger
+  , mkHLogger
+  , ExtraContext(..)
+  ) where
+
+import qualified Control.Retry              as R
+import qualified Data.Aeson                 as J
+import qualified Data.Aeson.Casing          as J
+import qualified Data.Aeson.TH              as J
+import qualified Data.ByteString.Lazy       as B
+import qualified Data.CaseInsensitive       as CI
+import qualified Data.TByteString           as TBS
+import qualified Data.Text                  as T
+import qualified Data.Text.Encoding         as TE
+import qualified Data.Text.Encoding.Error   as TE
+import qualified Data.Text.Lazy             as TL
+import qualified Data.Text.Lazy.Encoding    as TLE
+import qualified Data.Time.Clock            as Time
+import qualified Network.HTTP.Client        as H
+import qualified Network.HTTP.Types         as N
+import qualified Network.Wreq               as W
+import qualified Network.Wreq.Session       as WS
+import qualified System.Log.FastLogger      as FL
+
+import           Control.Exception          (try)
+import           Control.Lens
+import           Control.Monad.Except       (MonadError, throwError)
+import           Control.Monad.IO.Class     (MonadIO, liftIO)
+import           Control.Monad.Reader       (MonadReader, ask)
+import           Data.Has
+import           Hasura.Logging
+-- import           Data.Monoid
+import           Hasura.Prelude
+import           Hasura.RQL.Types.Subscribe
+
+-- import           Context                  (HTTPSessionMgr (..))
+-- import           Log
+
+type HLogger = (LogLevel, EngineLogType, J.Value) -> IO ()
+
+data ExtraContext
+  = ExtraContext
+  { elEventCreatedAt :: Time.UTCTime
+  , elEventId        :: TriggerId
+  } deriving (Show, Eq)
+
+$(J.deriveJSON (J.aesonDrop 2 J.snakeCase){J.omitNothingFields=True} ''ExtraContext)
+
+data HTTPSessionMgr
+  = HTTPSessionMgr
+  { _hsmSession         :: !WS.Session
```

Comment:
Get rid of this type. Just use WS.Session. Let's not support insecure connections to webhooks.